kategory-effects/src/main/kotlin/kategory/effects/data/DeferredKW.kt

```diff
@@ -0,0 +1,110 @@
+package kategory
+
+import kategory.effects.data.internal.Platform
+import kategory.effects.instances.DeferredKWInstances
+import kotlinx.coroutines.experimental.*
+import kotlin.coroutines.experimental.CoroutineContext
+
+typealias DeferredResult<A> = Either<Throwable, A>
+
+@higherkind data class DeferredKW<out A>(val coroutineContext: CoroutineContext, val thunk: (CoroutineContext) -> Deferred<A>) : DeferredKWKind<A> {
```

Comment:
All instances can be derived here with `@deriving`.

Reply:
pure requires a CoroutineContext to create an async block, which makes autogen fail :(